modules/fsi/src/interfacekernels/CoupledPenaltyInterfaceDiffusion.C

```diff
@@ -0,0 +1,111 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "CoupledPenaltyInterfaceDiffusion.h"
+
+registerMooseObject("FsiApp", CoupledPenaltyInterfaceDiffusion);
+
+InputParameters
+CoupledPenaltyInterfaceDiffusion::validParams()
```

Comment:
@somu15 I just came across this. Is there any reason that you name it with 'xxxDiffusion`? This seems to be a very generic  interface kernel to enforce continuity of any two variables. Other than diffusion problems, it can also enforce `displacement` variables for example. Maybe this can be put in framework. My 2 cents. Nice work!

Reply:
Thanks, @jiangwen84! @fdkong included this kernel. My thought is this kernel can consider any scalar variable and may require some modification to consider vector variables. Also, I am wondering whether this kernel can handle different physics across the interface.

Reply:
I told @ke7kto just the other day that I think `PenaltyInterfaceDiffusion` from the `test` directory should be moved up into the framework. This object looks like it works similarly. I agree that something like this should go into the framework